move/axelar_gateway/sources/gateway.move

```diff
@@ -71,11 +69,13 @@ public struct Gateway has key {
 
 /// The Status of the message.
 /// Can be either one of three statuses:
-/// - Non-existent: Set to bytes32(0)
+/// - NonExistent
 /// - Approved: Set to the hash of the message
-/// - Executed: Set to bytes32(1)
-public struct MessageStatus has store {
-    status: Bytes32,
+/// - Execute
+public enum MessageStatus has copy, drop, store {
+    NonExistent,
```

Comment:
non existent isn't needed since it corresponds to message not being in the store